lib/parser/parse-tree.h

```diff
@@ -2497,7 +2497,7 @@ WRAPPER_CLASS(FileUnitNumber, ScalarIntExpr);
 // R1203 internal-file-variable -> char-variable
 struct IoUnit {
   UNION_CLASS_BOILERPLATE(IoUnit);
-  std::variant<FileUnitNumber, Star, CharVariable> u;
+  std::variant<CharVariable, FileUnitNumber, Star> u;
```

Comment:
It seems misleading to use CharVariable here since any Variable will end up as a CharVariable. There's no enforcement as with DefaultCharExpr, for example.

So I think it would be clearer to eliminate CharVariable and just use Variable here.

Reply:
`CharVariable`  is a wrapper around `Variable` that will remain a `CharVariable` only if it is a character variable.

Reply:
I don't understand this. When does it become something other than a CharVariable?

Reply:
I'm fixing it in a parse tree rewriter.  More to come.

Reply:
As a data point, this code plus my code still has the original problem.